.github/workflows/clang-format-lint.yml

```diff
@@ -1,22 +1,34 @@
 name: Clang Format Lint
+
+on:
+  push:
+    branches:
+      - main
+      - rel-*
+  pull_request:
+    branches:
+      - main
+      - rel-*
+  workflow_dispatch:
+
 concurrency:
-  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
+  group: ${{ github.workflow }}-${{ github.event_name == 'pull_request' && github.ref
+    || github.sha }}
   cancel-in-progress: true
-on:
-  pull_request: { }
 
 jobs:
-  build:
-    runs-on: ubuntu-latest
-
+  lint-cpp:
+    runs-on:
+      - self-hosted
+      - "1ES.Pool=onnxruntime-genai-Ubuntu2204-AMD-CPU"
     steps:
-      - uses: actions/checkout@v2
-      - uses: actions/setup-python@v5
+      - name: Checkout code
+        uses: actions/checkout@v4
         with:
-          python-version: '3.11.x'
-          architecture: 'x64'
-      - uses: DoozyX/clang-format-lint-action@v0.17
+          submodules: 'false'
+
+      - name: Check format
```

Comment:
Should be configure lintrunner to support auto format and more linters etc?

Reply:
Just the same config in ort. Developers can run lintrunner f to format locally, a github bot can suggest fixes as well on the github UI.

Reply:
I am trying to change the C++ linter to a customized one for both repos. Now it does not have the ability to suggest fixes yet. It can be run locally. I plan to add more checker there, such as enabling clang-tidy, check if any C/C++ has an execution bit, etc.  And we will continue using lintrunner for python code. Is it ok?

Reply:
Sounds good to me! Question: will the clang-format component be the same? Maybe we can retain clang-format so that the local auto format experience is kept?

Reply:
They use the same config file. So it should be the same. However, the one used in ORT's CI pipeline was fetched from pypi.org, and the package didn't seem to be an official package from LLVM.  https://pypi.org/project/clang-format/
And, we want to reduce the usages of 3rd-party Github Actions. That's the main reason why I am doing these changes.

Reply:
It's ok to use the unofficial binaries locally, but our build pipelines need to hold a higher security bar.

Reply:
pytorch uses this https://github.com/pytorch/pytorch/blob/ff075d08150ec060f035b660789ebf1de8deb75b/.lintrunner.toml#L100-L105 to fetch a prebuilt, controlled binary. I wonder if we can do the same?